quattor/types/hardware.pan

```diff
@@ -75,25 +75,59 @@ type structure_bmc = {
 };
 
 @documentation{
+    PCI addresses
+}
+type structure_pci = {
+    "vendor" ? long
+    "device" ? long
+    "class"  ? long
+};
+
+@docmentation{
+    IB hardware address
+}
+type ibhwaddr = string with is_ibhwaddr (SELF);
+
+@documentation{
+    IB HCA
+}
+type structure_ibhca = {
+    include structure_annotation
+    "driver" ? string
+    "media" ? string
+    "name" ? string
+    "pxe" ? boolean
+    "boot" ? boolean
+    "ca_name" ? string
+    "portnum" ? long(1..)
+    "hwaddr" ? ibhwaddr
+    "active" ? boolean
+    "pci" ? structure_pci
+    "guid" ? string with is_ibguid (SELF)
+};
+
+@documentation{
     Card and/or addon
 }
 type structure_cards = {
-    # Indexed by device name (eth0, venet0...)
+    @{ desc =  Indexed by device name (eth0, venet0...)}
```

Comment:
what is the advantage of the `desc = `? these are not functions and we do not use this anywhere else in attribute annotation